packages/core/admin/shared/contracts/audit-logs.ts

```diff
@@ -0,0 +1,43 @@
+import { errors } from '@strapi/utils';
+import { Entity, SanitizedAdminUser } from './shared';
+import { Schema } from '@strapi/types';
+
+interface CollectionAuditLog extends Schema.CollectionType {
+  user?: SanitizedAdminUser;
+}
+
+interface SingleAuditLog extends Schema.SingleType {
+  user?: SanitizedAdminUser;
+}
```

Comment:
I'm not sure this is right, from what I see in my API response the audit log shape is very simple: only id, action, date, payload and user keys. Not attributes object or anything else

Also I think User is mandato

Reply:
So what would that look like?

```ts
interface AuditLog {
	id: Entity.ID;
	date: string;
	action: string; // is this a union of specific actions?
	payload: unknown;
	user: SanitizedAdminUser
}
```

What is a `user key`? and what is the `payload? 🤔

Reply:
Yes that looks good.

- user is the sanitized object 👍 it's now effectively required, but it wasn't when we launched the feature, so I think we have to leave it as optional
- action: currently we know the list so a union would work. But [we planned](https://github.com/strapi/strapi/blob/main/packages/core/admin/ee/server/services/audit-logs.js#L88-L89) on later letting users save custom events too. So I would leave it as a string
- payload depends on each action type, so I think unknown makes sense.

Reply:
we do know that payload is json actually

Reply:
Okay, let me make some changes and you can review again :)